hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineNotFoundException.java

```diff
@@ -18,18 +18,18 @@
 
 package org.apache.hadoop.hdds.scm.pipeline;
 
-import java.io.IOException;
+import org.apache.hadoop.hdds.scm.exceptions.SCMException;
 
 /**
  * Signals that a pipeline is missing from PipelineManager.
  */
-public class PipelineNotFoundException extends IOException{
+public class PipelineNotFoundException extends SCMException {
   /**
    * Constructs an {@code PipelineNotFoundException} with {@code null}
    * as its error detail message.
    */
   public PipelineNotFoundException() {
-    super();
+    super(ResultCodes.PIPELINE_NOT_FOUND);
```

Comment:
Looks like this ResultCode is lost when received at client.
But I see the message with whole stacktrace with debug.

Reply:
The latest commit adds the result code in the protobuf. Did you see that behavior in latest commit @bharatviswa504 ?

Reply:
Yes. I believe the ResultCodes in proto will help when we want to return ResultCodes in Response and construct the Exception based on that.

I believe when Rpc Layer constructs this Exception it used IOException default constructor and the ResultCode will be lost, and the message will be present.

Reply:
RemoteException unwrapException code and instantiateException, only message will be saved.
```public IOException unwrapRemoteException() {
    try {
      Class<?> realClass = Class.forName(getClassName());
      return instantiateException(realClass.asSubclass(IOException.class));
    } catch(Exception e) {
      // cannot instantiate the original exception, just return this
    }
    return this;
  }

private IOException instantiateException(Class<? extends IOException> cls)
      throws Exception {
    Constructor<? extends IOException> cn = cls.getConstructor(String.class);
    cn.setAccessible(true);
    IOException ex = cn.newInstance(this.getMessage());
    ex.initCause(this);
    return ex;
  }
```